### DISCUSSION TITLE
Motion Sensor Support

### DISCUSSION THREAD

**Discussion by evanbailey** [nirnachmani/Pixie-Plus-for-Home-Assistant#16] [Q&A]
Hi All,

I stupidly purchased one of the Pixie motion sensors (and buttons) without checking for HA compatibility. It works fine on it's own but I want to also trigger some Tapo smart plugs on motion. [Sensor is this one](https://www.sparkydirect.com.au/p/sal-sms862wf-wh-btam-smart-pixie-pir-sensor-outdoor-110-ip66?srsltid=AfmBOorpAlFnVcCLEYGC9rNSQoXE_fOM98n6HMR06HyAlJRksQPYmBz3)

Is there any way to make this work? Or any suggest on where I could start tweaking the code to add support for this?

Cheers!

**Comment by nirnachmani** [comment]
You will need to monitor the communication between the sensor and the app and then update the code with a new type of entity. It is not simple and will require much more than tweaking the code. 

You can ask the Pixie Plus team to send me the sensor so I can add it to the integration - they have done that before with other hardware.

A workaround will be to create an automation which will turns on the Tapo plugs when the Pixie light turns on (from the sensor and otherwise).

**Comment by evanbailey** [comment]
I'm happy to buy one and have it sent to you, and flick you some $ for a 6 pack + postage to send it back to me once you've done the updates. Would that work for you?

**Reply by nirnachmani** [reply]
Sorry for the late reply. I would prefer not to because I am not sure when I will have time to work on it, it may be only in few months.

**Comment by nirnachmani** [comment]
@evanbailey , I know it has been a while but if you want I can try to add support for the sensor. I will do it for the new local integration so it is best to move to it - https://github.com/nirnachmani/Pixie_Plus_Local

To make things easy, I uploaded a script that captures the required data so I can add it to the integration. 

The script is called capture_device_actions.py, I uploaded it to the repository.

It doesn't collect any sensitive information, only lists of devices, their parameters, and actions. It will create a file that you can review. It doesn't send anything automatically.

What it needs:

Python packages:
httpx
pycryptodome

Install them with:
```
pip install httpx pycryptodome
```
How to use it:

Run the script.
```
python3 capture_device_actions.py
```

-  Enter your Pixie Plus username and password when prompted.
-  Wait for it to discover the hub and build the device inventory.
-  Pick a device by ID from the table.
-  Enter the official Pixie Plus product details for that device:
  -  official product name (from the package)
  - official model (from the package)
  - device description/type
- Enter an action label describing what you are about to do.
- Press Enter to start capture, then physically perform the action in the app/device.
- Repeat for more actions on that device, or go back and choose another device. Best to capture all possible actions.
- Type done when finished.
- Review the generated text report and send it.
